### DISCUSSION TITLE
How can I use the context during a subscription.trigger?

### DISCUSSION THREAD

**Discussion by dominikklein** [rmosolgo/graphql-ruby#5099] [Q&A]
The subscription trigger has the following parameters:

```
#trigger(event_name, args, object, scope: nil, context: {}) ⇒ void
```

What is the use case of the context here? In my situation, I want to handover some additional context to the possible executed subscription?

**Comment by rmosolgo** [comment]
Hi! Good question. I'm honestly not sure what that'd come in handy for. Here's the source: https://github.com/rmosolgo/graphql-ruby/blob/a814873ce391f9031bbb6465edf02c13d419e33f/lib/graphql/subscriptions.rb#L60-L93

It looks like there, it would be used for finding the field that the trigger refers to, so it would be passed to `visible?` methods. 

It's also passed to `Subscriptions::Event.new`, so its used a few places there: 

https://github.com/rmosolgo/graphql-ruby/blob/a814873ce391f9031bbb6465edf02c13d419e33f/lib/graphql/subscriptions/event.rb#L42C27-L42C41

https://github.com/rmosolgo/graphql-ruby/blob/a814873ce391f9031bbb6465edf02c13d419e33f/lib/graphql/subscriptions/event.rb#L36

And I think this is a red herring, it's used when running the _initial_ subscription, but not used for subsequent updates: 

https://github.com/rmosolgo/graphql-ruby/blob/a814873ce391f9031bbb6465edf02c13d419e33f/lib/graphql/subscriptions/event.rb#L46-L60

Is there something particular you're hoping to use it for? Or something you tried but didn't work?

**Reply by dominikklein** [reply]
Our use case is that we have a subscription that should not be triggered when the "own" browser tab is using the mutation. We tried to have a unique id for the browser tab. But failed to have a handover of this id during the triggering.
